packages/synthetics-sdk-broken-links/src/broken_links_main.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import puppeteer from 'puppeteer';
+import {
+  getRuntimeMetadata,
+  SyntheticResult,
+} from '@google-cloud/synthetics-sdk-api';
+import {
+  closeBrowser,
+  createSyntheticResult,
+  LinkIntermediate,
+  openNewPage,
+  setDefaultOptions,
+  shuffleAndTruncate,
+  validateInputOptions,
+} from './link_utils';
+import {
+  checkLink,
+  checkLinks,
+  retrieveLinksFromPage,
+  getGenericSyntheticResult,
+} from './broken_links_func';
+
+export interface BrokenLinkCheckerOptions {
+  origin_url: string;
+  link_limit?: number;
+  query_selector_all?: string;
+  get_attributes?: string[];
+  link_order?: LinkOrder;
+  link_timeout_millis?: number | undefined;
+  max_retries?: number | undefined;
+  max_redirects?: number | undefined;
+  wait_for_selector?: string;
+  per_link_options?: { [key: string]: PerLinkOption };
+}
+
+export interface PerLinkOption {
+  link_timeout_millis?: number;
+  expected_status_code?: StatusClass | number;
+}
+
+export enum LinkOrder {
+  FIRST_N = 'FIRST_N',
+  RANDOM = 'RANDOM',
+}
+
+export enum StatusClass {
+  STATUS_CLASS_UNSPECIFIED = 'STATUS_CLASS_UNSPECIFIED',
+  STATUS_CLASS_1XX = 'STATUS_CLASS_1XX',
+  STATUS_CLASS_2XX = 'STATUS_CLASS_2XX',
+  STATUS_CLASS_3XX = 'STATUS_CLASS_3XX',
+  STATUS_CLASS_4XX = 'STATUS_CLASS_4XX',
+  STATUS_CLASS_5XX = 'STATUS_CLASS_5XX',
+  STATUS_CLASS_ANY = 'STATUS_CLASS_ANY',
+}
+
+export async function runBrokenLinks(
```

Comment:
It is not quite as simple as that but a lot of functionality has been broken down into separate helpers. So that the try block is more readable. 

I think now it is still clear what is happening without being as complex:

```
process inputs
checkOriginLink
    return if it failed
scrape links
check scraped links
return
```